### PR TITLE
Fix HTML parsing / chunking by using markdown conversion (for now)

### PR DESCRIPTION
**Pull request by oxaroky02**
As a quick fix, I'm using `pandoc` to convert HTML to markdown for chunking.

### CODE REVIEW

File: app/services/parsers/html_via_markdown.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      grep -v '<!-- -->' |\
+      sed -e ':again' -e N -e '$!b again' -e 's/{[^}]*}//g'
+    TEXT
+
+    ENDFRONTMATTER = "---\n\n"
+
+    def initialize(document)
+      super(document)
+      # Convert the HTML into markdown, strip out all the junk we don't need
+      rawtxt, serr, status = Open3.capture3(CMD, stdin_data: @document.contents, binmode: true)
```
Comment (nickthecook) [inline app/services/parsers/html_via_markdown.rb:18]: Smart, saving that extra character in the variable name. ;)

Comment (nickthecook) [inline app/services/parsers/html_via_markdown.rb:18]:
```suggestion
      r, e, s = Open3.capture3(CMD, stdin_data: @document.contents, binmode: true)
```

File: app/services/parsers/html_via_markdown.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      if status.success?
+        # extract the front matter from the raw text
+        @title = rawtxt.match(/title: (.+)\n/).captures.join
+        # remove the front matter to get the body
+        @text = rawtxt[(rawtxt.index("---\n\n") + 5)..]
+      else
+        Rails.logger.error("Error running '#{cmd}' on HTML: #{@document.filename}\n#{serr}")
+        raise StandardError, "Error converting HTML to markdown: #{@document.filename}'"
```
Comment (nickthecook) [inline app/services/parsers/html_via_markdown.rb:26]: I imagine we don't want the full `stderr` here, but is there a predictable format to the errors such that you could add the first line of stderr and get something descriptive?

Comment (nickthecook) [reply app/services/parsers/html_via_markdown.rb:26]: Actually, this should probably `raise` so the job fails.

Comment (oxaroky02) [reply app/services/parsers/html_via_markdown.rb]: Fixed.

**Pull request merged**